### PR TITLE
chore(Github Action|CI): add Github Action for testing

### PR DESCRIPTION
**Pull request by MichelDiz**
Reusing existing test scripts to bring CI into the repository.

- - -
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph-js/169)

### CODE REVIEW

**Comment by skrdgraph** [comment]
@joshua-goldstein / @dshekhar95 could you also take a look .. I am starting to review this. If you see any issues pls flag it here. I know you both are fixing other clients as well (hence adding you)

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
```
Comment (adityasadalage) [inline .github/workflows/node.js.yml]: Change name of file to "ci-dgraph-js.yml" to bring in line with CI runs in other dgraph repos

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
```
Comment (adityasadalage) [inline .github/workflows/node.js.yml]: name should be ci-dgraph-js

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
```
Comment (adityasadalage) [inline .github/workflows/node.js.yml]: Need to add the following PR types

Comment (adityasadalage) [reply .github/workflows/node.js.yml]: pull_request:
    types:
      - opened
      - reopened
      - synchronize
      - ready_for_review
    branches:
      - master

**Review by dshekhar95 — APPROVED** [review]
Looks good Michel. few thoughts

1. Not a breaking issue here, but need to have some discussion on standardizing node versions across our projects

2. master vs main?
3. Explore some flags to add to the npm test command (coverage, watchAll, CI, testresultsprocessor)

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - name: Install Dependencies
+      run: |
+        chmod +x ./scripts/build.sh
+        chmod +x ./scripts/install_dgraph.sh
+        bash ./scripts/install_dgraph.sh
+    - run: npm ci --legacy-peer-deps
+    - run: ./scripts/build.sh
```
Comment (adityasadalage) [inline .github/workflows/node.js.yml]: `name: run tests` could be added here

**Review by adityasadalage — APPROVED** [review]

**Comment by MichelDiz** [comment]
@skrdgraph 

Good, all points we should address with more discussion. 
Standarize Node is really good, but hard. Some repos we use legacy code, so we need to review then and replace if necessary.
Ratel is an example, it is painful to make any progress there.

Change it to main we will need to check the side effects. If you have something in another repository being master instead of main can break things. So we need more ppl looking into it.

coverage, watchAll, CI, testresultsprocessor. Again, other stuff we can add to Jira and work on.

File: .github/workflows/node.js.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    branches: [ "master" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x, 19.x]
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: is there a good reason to support multiple node versions?  my suggestion would be to stick to one version for now and get everything working, and then if we want to we can revisit this and add more versions.

File: .github/workflows/node.js.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - name: Install Dependencies
+      run: |
+        chmod +x ./scripts/build.sh
+        chmod +x ./scripts/install_dgraph.sh
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: not necessary if invoking script with `bash`

File: .github/workflows/node.js.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - name: Install Dependencies
+      run: |
+        chmod +x ./scripts/build.sh
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: not necessary if invoking script with `bash`

File: .github/workflows/node.js.yml (hunk 4 above)
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: change to `run: bash ./scripts/build.sh`

File: .github/workflows/node.js.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: change to `ubuntu-20.04`

**Review by joshua-goldstein — CHANGES_REQUESTED** [review]

File: .github/workflows/node.js.yml (hunk 5 above)
Comment (MichelDiz) [reply .github/workflows/node.js.yml]: In the past it was a practice of the old repo maintainer. He tested on old versions in Team City. Until someone come and tell him to stop supporting legacy. The argument is that many users use legacy libraries and run on a legacy node binary. If we stop testing we don't guarantee support for them.

18.x is Active LTS and will be dropped in 2025-04-30

https://en.wikipedia.org/wiki/Node.js#Releases

File: .github/workflows/node.js.yml (hunk 7 above)
Comment (MichelDiz) [reply .github/workflows/node.js.yml]: I'm going to remove them all and just use bash. I was using it like this out of habit.

**Review by joshua-goldstein — APPROVED** [review]

**Pull request merged**